### PR TITLE
Remove timeout from advanced security scan

### PR DESCRIPTION
**Pull request by Or-Geva**
- [ ] All [tests](https://github.com/jfrog/jfrog-vscode-extension#building-and-testing-the-sources) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] I used `npm run format` for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: src/main/utils/scanUtils.ts [hunk 1]
```diff
@@ -300,10 +300,4 @@ export class FileScanError extends Error {
...
 export class ScanCancellationError extends Error {
     message: string = 'Scan was cancelled';
-}
-
-export class ScanTimeoutError extends Error {
-    constructor(scan: string, public time_millisecs: number) {
-        super(`Task ${scan} timed out after ${time_millisecs}ms`);
-    }
-}
+}
```
Comment (yahavi) [inline src/main/utils/scanUtils.ts:303]: Let's run `npm run format`

File: src/main/utils/runUtils.ts [hunk 2]
```diff
@@ -1,53 +1,4 @@
...
-     * Creates a Promise that is resolved with an array of results when all of the provided Promises resolve, or rejected when:
-     * 1. Any Promise is rejected.
-     * 2. Cancel was requested.
-     * 3. Timeout reached.
-     * @param timeout - time in millisecs until execution timeout
-     * @param checkCancel - check if cancel was requested
-     * @param tasks - the promises that the new promise will wrap
-     * @returns Promise that wrap the given promises
-     */
-    static async runWithTimeout<T>(timeout: number, checkCancel: () => void, ...tasks: (Promise<T> | Task<T>)[]): Promise<T[]> {
```
Comment (yahavi) [inline src/main/utils/runUtils.ts:22]: I think it is important to kill the process if the user clicks on `Cancel`

**Review by yahavi — CHANGES_REQUESTED** [review]

File: src/main/utils/scanUtils.ts [hunk 3]
```diff
@@ -158,8 +159,62 @@ export class ScanUtils {
...
+        command: string,
+        checkCancel: () => void,
+        cwd?: string,
+        env?: NodeJS.ProcessEnv | undefined
+    ): Promise<string> {
+        return new Promise<string>((resolve, reject) => {
+            try {
+                const childProcess: exec.ChildProcess = exec.exec(
+                    command,
+                    { cwd: cwd, maxBuffer: ScanUtils.SPAWN_PROCESS_BUFFER_SIZE, env: env },
```
Comment (yahavi) [inline src/main/utils/scanUtils.ts:180]:
```suggestion
                    { cwd: cwd, maxBuffer: ScanUtils.SPAWN_PROCESS_BUFFER_SIZE, env: env } as exec.ExecOptions,
```

File: src/main/utils/scanUtils.ts [hunk 4]
```diff
@@ -158,8 +159,62 @@ export class ScanUtils {
...
+
+    /**
+     * Cancels the child process if cancellation is requested.
+     * @param childProcess - The child process to be cancelled.
+     * @param checkCancel - A function to check if cancellation is requested.
+     * @param reject - A function to reject the promise.
+     */
+    private static cancelProcess(childProcess: exec.ChildProcess, checkCancel: () => void, reject: (reason?: any) => void): void {
+        try {
+            checkCancel();
```
Comment (yahavi) [inline src/main/utils/scanUtils.ts:208]: You already kill the process. Why running checkCancel?
Moreover - I think that running checkCancel without killing the process may end with a Zombie process.

File: src/test/tests/scanUtils.test.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+        afterEach(() => {
+            clearIntervalSpy.restore();
+        });
+
+        const dummyCheckError: () => void = () => {
+            return;
+        };
+
+        it('should execute a command successfully', async () => {
```
Comment (yahavi) [inline src/test/tests/scanUtils.test.ts:23]: Let's use the same project conventions for the test names, for example in this case:
```suggestion
        it('Execute a command successfully', async () => {
```

File: src/main/utils/scanUtils.ts (hunk 4 above)
Comment (Or-Geva) [reply src/main/utils/scanUtils.ts:208]: Rename to `cancelIfRequested`

**Pull request merged**